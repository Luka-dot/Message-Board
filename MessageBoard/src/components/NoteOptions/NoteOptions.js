import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types'

const NoteOptions = ({ handleType, deleteNote, handleClose, type, displayError, changeColor, title }) => {

  return (
    <div className="Note-Options">
      <div className="Color-Picker">
        <button className="Color Yellow" name="#FFE680" onClick={(e) => changeColor(e)}></button>
        <button className="Color Green" name="#D6FFB7" onClick={(e) => changeColor(e)}></button>
        <button className="Color Purple" name="#EEF5DB" onClick={(e) => changeColor(e)}></button>
      </div>
      <div className="Note-SubOptions">
      {
        title.length < 1 || !displayError &&
        <button className="Note-Save" onClick={handleType} type="submit">
          <i className="fas fa-save"></i>
        </button>
      }
      {
        type === "existing-note" &&
          <button className="Note-Delete" onClick={deleteNote}>
            <i className="fas fa-trash-alt"></i>
          </button>
      }
      <NavLink onClick={handleClose} to="/" className="Note-Close">
        <button>Close</button>
      </NavLink>
      </div>
    </div>
  )
}

NoteOptions.propTypes = {
  handleType: PropTypes.func.isRequired,
  deleteNote: PropTypes.func.isRequired,
  handleClose: PropTypes.func.isRequired,
  changeColor: PropTypes.func.isRequired
}

export default NoteOptions;
