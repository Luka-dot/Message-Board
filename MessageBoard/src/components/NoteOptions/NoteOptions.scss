.Note-Options {
  display: flex;
  justify-content: space-between;
  padding: .5em;
  margin-top: 1em;
  box-shadow: 0 -2px 5px rgba(0,0,0,0.2);
  align-items: flex-end;

  .Color-Picker {
    display: flex;
    button {
      outline: none;
      background: #231f20;
      display: block;
      width: 20px;
      border-radius: 20px;
      height: 20px;
      margin-right: 10px;
      cursor: pointer;
      box-shadow: 0px 0px 5px #00000038;

      &.Yellow {background: #FFE680}
      &.Green {background: #D6FFB7}
      &.Purple {background: #EEF5DB}
    }
  }

  .Note-SubOptions {

      button {
        background: none;
        outline: none;
        border: none;
        font-size: 1.2em;
        cursor: pointer;
        color: #6f6f6f;
        opacity: 1;
        transition: all .5s ease;
        margin-right: .5em;

        &:hover {
          opacity: .6;
        }
      }

      .Note-Save {
        i {
          position: relative;
          bottom: -3px;
        }
      }

      .Note-Delete {
        i {
          position: relative;
          bottom: -2px;
        }
      }

      .Note-Close {

        button {
          margin-right: 0;
          font-family: $font-Roboto;
          outline: none;
          cursor: pointer;
          border: none;
          height: 100%;
          box-sizing: border-box;
          color: #fff;
          background-color: #6f6f6f;
          overflow: hidden;
          text-overflow: ellipsis;
          white-space: nowrap;
          letter-spacing: .01785714em;
          font-size: .75rem;
          font-weight: 500;
          line-height: .6;
          height: 24px;
          padding: 8px 15px;
          border-radius: 4px;
        }
      }

  }


}
