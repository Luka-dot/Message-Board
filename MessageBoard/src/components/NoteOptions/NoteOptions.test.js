import React from 'react';
import { shallow } from 'enzyme';
import NoteOptions from './NoteOptions';


describe('NoteOptions', () => {
  let wrapper;

  const mockHandleType = jest.fn()
  const mockDeleteNote = jest.fn()
  const mockHandleClose = jest.fn()
  const mockChangeColor = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<NoteOptions handleType={mockHandleType}
                                   deleteNote={mockDeleteNote}
                                   handleClose={mockHandleClose}
                                   changeColor={mockChangeColor}
                                   type="existing-note"
                                   title="A Title"
                                  />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call handleType when the save button is clicked', () => {
    wrapper.find('.Note-Save').simulate('click')
    expect(mockHandleType).toHaveBeenCalled()
  })

  it('should call deleteNote when the delete button is clicked', () => {
    wrapper.find('.Note-Delete').simulate('click')
    expect(mockDeleteNote).toHaveBeenCalled()
  })

  it('should call handleClose when the close note button is clicked', () => {
    wrapper.find('.Note-Close').simulate('click')
    expect(mockHandleClose).toHaveBeenCalled()
  })

  it('should call changeColor when the yellow color button is clicked', () => {
    wrapper.find('.Color.Yellow').simulate('click')
    expect(mockChangeColor).toHaveBeenCalled()
  })

  it('should call changeColor when the yellow color button is clicked', () => {
    wrapper.find('.Color.Yellow').simulate('click')
    expect(mockChangeColor).toHaveBeenCalled()
  })

  it('should call changeColor when the green color button is clicked', () => {
    wrapper.find('.Color.Green').simulate('click')
    expect(mockChangeColor).toHaveBeenCalled()
  })

  it('should call changeColor when the purple color button is clicked', () => {
    wrapper.find('.Color.Purple').simulate('click')
    expect(mockChangeColor).toHaveBeenCalled()
  })

})
