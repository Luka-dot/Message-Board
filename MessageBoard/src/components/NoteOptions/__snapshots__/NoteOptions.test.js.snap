// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteOptions should match the snapshot 1`] = `
<div
  className="Note-Options"
>
  <div
    className="Color-Picker"
  >
    <button
      className="Color Yellow"
      name="#FFE680"
      onClick={[Function]}
    />
    <button
      className="Color Green"
      name="#D6FFB7"
      onClick={[Function]}
    />
    <button
      className="Color Purple"
      name="#EEF5DB"
      onClick={[Function]}
    />
  </div>
  <div
    className="Note-SubOptions"
  >
    <button
      className="Note-Save"
      onClick={[MockFunction]}
      type="submit"
    >
      <i
        className="fas fa-save"
      />
    </button>
    <button
      className="Note-Delete"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-trash-alt"
      />
    </button>
    <NavLink
      className="Note-Close"
      onClick={[MockFunction]}
      to="/"
    >
      <button>
        Close
      </button>
    </NavLink>
  </div>
</div>
`;
